ios/RNSBarButtonItem.mm

```diff
@@ -0,0 +1,269 @@
+  if (sfSymbolName) {
+    self.image = [UIImage systemImageNamed:sfSymbolName];
+  }
+
+  NSDictionary *titleStyle = dict[@"titleStyle"];
+  if (titleStyle) {
+    [self setTitleStyle:titleStyle];
+  }
+
+  id tintColorObj = dict[@"tintColor"];
+  if (tintColorObj) {
+    self.tintColor = [RCTConvert UIColor:tintColorObj];
+  }
+
+#if RNS_IPHONE_OS_VERSION_AVAILABLE(16_0)
+  if (@available(iOS 16.0, *)) {
+    NSNumber *hiddenNum = dict[@"hidden"];
+    if (hiddenNum != nil) {
+      self.hidden = [hiddenNum boolValue];
+    }
+  }
+#endif
+
+  NSNumber *selectedNum = dict[@"selected"];
+  if (selectedNum != nil) {
+    self.selected = [selectedNum boolValue];
+  }
+
+  NSNumber *enabledNum = dict[@"enabled"];
+  if (enabledNum != nil) {
+    self.enabled = [enabledNum boolValue];
+  }
+
+  NSNumber *width = dict[@"width"];
+  if (width) {
+    self.width = [width doubleValue];
+  }
+
+  NSNumber *changesSelectionAsPrimaryActionNum = dict[@"changesSelectionAsPrimaryAction"];
+  if (changesSelectionAsPrimaryActionNum != nil) {
+    self.changesSelectionAsPrimaryAction = [changesSelectionAsPrimaryActionNum boolValue];
+  }
+
+#if RNS_IPHONE_OS_VERSION_AVAILABLE(26_0)
+  if (@available(iOS 26.0, *)) {
+    NSNumber *hidesSharedBackgroundNum = dict[@"hidesSharedBackground"];
+    if (hidesSharedBackgroundNum != nil) {
+      self.hidesSharedBackground = [hidesSharedBackgroundNum boolValue];
+    }
+    NSNumber *sharesBackgroundNum = dict[@"sharesBackground"];
+    if (sharesBackgroundNum != nil) {
+      self.sharesBackground = [sharesBackgroundNum boolValue];
+    }
+    NSString *identifier = dict[@"identifier"];
+    if (identifier != nil) {
+      self.identifier = identifier;
+    }
+    NSDictionary *badgeObj = dict[@"badge"];
+    if (badgeObj != nil) {
+      [self createBadge:badgeObj];
+    }
+  }
+#endif
+
+  NSString *style = dict[@"style"];
+  if (style) {
+    if ([style isEqualToString:@"done"]) {
+      self.style = UIBarButtonItemStyleDone;
+    } else if ([style isEqualToString:@"prominent"]) {
+#if RNS_IPHONE_OS_VERSION_AVAILABLE(26_0)
+      if (@available(iOS 26.0, *)) {
+        self.style = UIBarButtonItemStyleProminent;
+      }
+#endif
+    } else {
+      self.style = UIBarButtonItemStylePlain;
+    }
+  }
+
+  if (dict[@"accessibilityLabel"]) {
+    self.accessibilityLabel = dict[@"accessibilityLabel"];
+  }
+  if (dict[@"accessibilityHint"]) {
+    self.accessibilityHint = dict[@"accessibilityHint"];
+  }
+
+  NSDictionary *menu = dict[@"menu"];
+  if (menu) {
+    self.menu = [[self class] initUIMenuWithDict:menu menuAction:menuAction];
+  }
+
+  NSString *buttonId = dict[@"buttonId"];
+  if (buttonId && action) {
+    self.target = self;
+    self.action = @selector(handleBarButtonItemPress:);
+    objc_setAssociatedObject(self, &RNSBarButtonItemIdKey, buttonId, OBJC_ASSOCIATION_COPY_NONATOMIC);
+    objc_setAssociatedObject(self, &RNSBarButtonItemActionKey, action, OBJC_ASSOCIATION_COPY_NONATOMIC);
```

Comment:
Why do we go for ObjC runtime functions instead of regular members? What's the reasoning here?